contractcourt/channel_arbitrator.go

```diff
@@ -1430,6 +1498,18 @@ func (c *ChannelArbitrator) resolveContract(currentContract ContractResolver) {
 						"contract: %v", err)
 				}
 
+				// Swap contract in resolvers list. This is to
+				// make sure that reports are queried from the
+				// new resolver.
+				c.activeResolversLock.Lock()
+				for i, r := range c.activeResolvers {
+					if r == currentContract {
```

Comment:
Comparing pointers makes me nervous at times, instead this can use the `ResolverKey()` method directly.

Reply:
Yes valid point. Fixed and replace operation extract to method.